docs/_posts/examples/3400-01-31-animate-a-line.html

```diff
@@ -0,0 +1,122 @@
+--- 
+layout: example 
+category: example 
+title: Animate a line 
+description: Animate a line by updating a GeoJSON source on each frame. 
+tags: 
+ - layers 
+ - sources 
+---
+<style>
+button {
+  position: absolute;
+  margin: 20px;
+}
+</style>
+<div id='map'></div>
+<button id='replay'>Replay</button>
+<script>
+var map = new mapboxgl.Map({
+  container: 'map',
+  style: 'mapbox://styles/mapbox/streets-v8',
+  center: [-180, 0],
+  zoom: 0.5
+});
+
+// Create a GeoJSON source with an empty lineString.
+var geojson = {
+  "type": "FeatureCollection",
+  "features": [{
+    "type": "Feature",
+    "geometry": {
+      "type": "LineString",
+      "coordinates": [
+        [0, 0]
+      ]
+    }
+  }]
+};
+
+map.on('load', function() {
+
+  // Add a source and layer displaying the line which will be
+  // animated in a circle as a sine wave along the map.
+  map.addSource('line-animation', {
+    'type': 'geojson',
+    'data': geojson
+  });
+
+  map.addLayer({
+    'id': 'line-animation',
+    'type': 'line',
+    'source': 'line-animation',
+    'layout': {
+      'line-cap': 'round',
+      'line-join': 'round'
+    },
+    'paint': {
+      'line-color': '#ed6498',
+      'line-width': 5,
+      'line-opacity': .8
+    }
+  });
+
+  var speedFactor = 40; // number of frames per longitude degree
+  var animation; // to store the animation
+  var startTime = performance.now(); // system timestamp
+  var progress; // progress = timestamp - startTime
+
+  animateLine();
+
+  document.getElementById("replay").addEventListener("click", function() {
+    // in case there is an ongoing animation
+    cancelAnimationFrame(animation);
+    startTime = performance.now();
+    geojson = {
+      "type": "FeatureCollection",
+      "features": [{
+        "type": "Feature",
+        "geometry": {
+          "type": "LineString",
+          "coordinates": [
+            [0, 0]
+          ]
+        }
+      }]
+    };
+    // play animation
+    animateLine();
+  });
+
+  function animateLine(timestamp) {
+
+    // requestAnimationFrame pauses when the tab is inactive
+    // reset startTime and progress when switched back
+    document.addEventListener('visibilitychange', function(){
```

Comment:
this call shouldn't be inside of `animateLine`: `animateLine` will be called many times but `document.addEventListener` should be called once.